logstash-core/lib/logstash/plugins/builtin/pipeline/input.rb

```diff
@@ -68,9 +68,10 @@ def internalReceive(events)
         stream_position = stream_position + 1
       end)
       ReceiveResponse.completed()
-    rescue java.lang.InterruptedException, IOError => e
-      # maybe an IOException in enqueueing
-      logger.debug? && logger.debug('queueing event failed', message: e.message, exception: e.class, backtrace: e.backtrace)
+    rescue java.lang.InterruptedException, org.logstash.ackedqueue.QueueRuntimeException, IOError => e
+      # WARN is intentional, if ensure_delivery disabled we lose warnings in PipelineBus.java caused by ReceiveResponse.failed_at(stream_position, e)
+      logger.warn('queueing event failed', message: e.message, exception: e.class)
+      logger.debug? && logger.debug('queueing event failed trace', message: e.message, exception: e.class, backtrace: e.backtrace)
```

Comment:
Per the semantics of `ensure_delivered => false` and its need to be explicitly-configured by the sender, I am _not_ in favor of elevating missed deliveries to `WARN`-level logs.

We know that customers rely on this feature to configure upstream pipelines to send events to downstream ones if they exist and to tolerate missing pipelines, and we do not want to flood them with warnings when they have already explicitly asked for behaviour that tolerates downstream being missing.

```suggestion
      logger.debug? && logger.debug('queueing event failed', message: e.message, exception: e.class, backtrace: e.backtrace)
```